pkg/smokescreen/smokescreen.go

```diff
@@ -421,15 +421,74 @@ func hasPort(s string) bool {
 	return strings.LastIndex(s, "]") < strings.LastIndex(s, ":")
 }
 
-// normalizeHost returns host (as string) and port (as int) derived from the
+func _normalizePort(s string) (port int, err error) {
+	if strings.HasPrefix(s, "0") && s != "0" {
+		return 0, fmt.Errorf("invalid port number %s: decimal representation required", s)
+	}
+	port, err = strconv.Atoi(s)
+	if err != nil {
+		return 0, fmt.Errorf("invalid port number %q: %v", s, err)
+	}
+	if port < portMin || port > portMax {
+		return 0, fmt.Errorf("invalid port number %d: must be between %d and %d", port, portMin, portMax)
+	}
+	return
+}
+
+func _normalizeHost(s string, forceFQDN bool) (string, error) {
+	if ip := net.ParseIP(s); ip != nil {
+		// IP addresses might have different but equivalent representations
+		// (e.g., `2001:DB8::` and `2001:db8::` are the same address).
+		// This function provides a consistent representation.
+		return ip.String(), nil
+	}
+	// If it's not an IP address then it must be a domain name.
+	// Convert it to Punycode it so that we deal only with with ASCII from now on.
+	// This way we can find out whether the domain name is malformed.
+	domain, err := idna.Lookup.ToASCII(s)
+	if err != nil {
+		return "", fmt.Errorf("invalid domain '%v': %v", s, err)
```

Comment:
FYI you can also use the `%q` format string to wrap a value in quoates.

Reply:
Unfortunately double quotes will get escaped in the log output so it won't look as good.